### PR TITLE
feat: add types and docstrings for browse command

### PR DESCRIPTION
**Pull request by PaulFarault**
#### Which issue(s) this PR fixes

Fixes None

#### Additional comments

For the browse command:

- add types
- add docstrings
- rename internal functions to start with a `_`

#### Agreements

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp/cli/commands/browse.py [hunk 1]
```diff
@@ -161,17 +186,35 @@ def print_formatted_operation_log(operation_log):
...
 
-def translate_timezone(timestamp):
+def _translate_timezone(timestamp: datetime) -> datetime:
+    """Translate a timestamp from UTC to the local timezone."""
     return timestamp.replace(tzinfo=timezone.utc).astimezone(LOCAL_TIMEZONE)
 
 
-def format_service_component(component_version_log):
+def _format_service_component(component_version_log: ComponentVersionLog) -> str:
+    """Format a service component log into a string.
```
Comment (rpignolet) [inline tdp/cli/commands/browse.py:195]:
```suggestion
    """Format a component log into a string.
```

File: tdp/cli/commands/browse.py [hunk 2]
```diff
@@ -161,17 +186,35 @@ def print_formatted_operation_log(operation_log):
...
+
+    Args:
+        component_version_log: Service component log to format
+
+    Returns:
+        <service>_<component> string.
```
Comment (rpignolet) [inline tdp/cli/commands/browse.py:201]:
```suggestion
        <service>_<component> string or <service> string.
```

**Review by rpignolet — APPROVED** [review]

**Comment by mdrutel** [comment]
Hi,

When I try to display 1 successful deployment, I have this error :+1: 
```
(venv) $ tdp browse
2023-07-11 08:50:55,858 - DEBUG - tdp.<module> - Logger initialized
Error: _format_component_version_log() missing 1 required positional argument: 'headers'
```

File: tdp/cli/commands/browse.py [hunk 3]
```diff
@@ -161,17 +186,35 @@ def print_formatted_operation_log(operation_log):
...
 
 
-def translate_timezone(timestamp):
+def _translate_timezone(timestamp: datetime) -> datetime:
+    """Translate a timestamp from UTC to the local timezone."""
     return timestamp.replace(tzinfo=timezone.utc).astimezone(LOCAL_TIMEZONE)
 
 
-def format_service_component(component_version_log):
+def _format_service_component(component_version_log: ComponentVersionLog) -> str:
```
Comment (PaulFarault) [inline tdp/cli/commands/browse.py:194]:
```suggestion
def _format_component_name(component_version_log: ComponentVersionLog) -> str:
```

File: tdp/cli/commands/browse.py [hunk 4]
```diff
@@ -184,16 +227,16 @@ def custom_format(key, value):
         elif key == "component_version":
             if len(value) > 2:
                 return (
-                    format_service_component(value[0])
+                    _format_component_version_log(value[0])
```
Comment (mdrutel) [inline tdp/cli/commands/browse.py:230]: Is my error related to the fact tha there is no second argument ?

Comment (PaulFarault) [inline tdp/cli/commands/browse.py:230]:
```suggestion
                    _format_component_name(value[0])
```

File: tdp/cli/commands/browse.py [hunk 5]
```diff
@@ -184,16 +227,16 @@ def custom_format(key, value):
...
                     + ",...,"
-                    + format_service_component(value[-1])
+                    + _format_component_version_log(value[-1])
```
Comment (mdrutel) [inline tdp/cli/commands/browse.py:232]: Same thing

Comment (PaulFarault) [inline tdp/cli/commands/browse.py:232]:
```suggestion
                    + _format_component_name(value[-1])
```

File: tdp/cli/commands/browse.py [hunk 6]
```diff
@@ -184,16 +227,16 @@ def custom_format(key, value):
...
                 )
             return ",".join(
-                format_service_component(component_version_log)
+                _format_service_component(component_version_log)
```
Comment (PaulFarault) [inline tdp/cli/commands/browse.py:235]:
```suggestion
                _format_component_name(component_version_log)
```

**Review by mdrutel — APPROVED** [review]
Test is now OK :+1:

**Pull request merged**